crypto/x509/x509_vfy.c

```diff
@@ -131,7 +131,7 @@ static int lookup_cert_match(X509 **result, X509_STORE_CTX *ctx, X509 *x)
         xtmp = NULL;
     }
     ret = xtmp != NULL;
-    if (ret) {
+    if (ret && result != NULL) {
```

Comment:
This change is not needed, or is not what you had in mind.  The `result` output parameter (pointer) is never NULL, we'd have crashed at `*result = NULL` on line 119 if it were.

Reply:
Oh, right! Fixed that.
Also added a comment on the use of `result`:
```
 * If result is not NULL, on success *result will hold the up-ref'd found cert.
```
The idea is that the function can now also be used as follows:
```
static int directly_trusted(X509_STORE_CTX *ctx, X509 *cert)
{
    return X509_check_trust(cert, ctx->param->trust,
                            ctx->param->flags) != X509_TRUST_REJECTED
        && lookup_cert_match(NULL, ctx, cert) != 0;
}
```